internal/controller/apiextensions/composite/reconciler.go

```diff
@@ -593,6 +618,14 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 		return reconcile.Result{Requeue: true}, errors.Wrap(r.client.Status().Update(ctx, xr), errUpdateStatus)
 	}
 
+	if r.kindObserver != nil {
+		var gvks []schema.GroupVersionKind
+		for _, ref := range xr.GetResourceReferences() {
+			gvks = append(gvks, ref.GroupVersionKind())
+		}
+		r.kindObserver.Watch(gvks...)
```

Comment:
What if a GVK appears in the resource references but then is later removed? This could happen if the XR updates to a CompositionRevision that no longer produces a resource with that GVK. Do we need to stop the watch?

Reply:
There is another reconcile loop running every 10s removing old GVKs. Originally, I also started informer that, but changed that. So maybe we can reduce 10s to 30s or a minute or so.

Reply:
Changed to 1m.

Reply:
Is it worth noting on the `Watch` interface that the implementation is responsible for garbage collecting watches that are no longer needed?